src/towncrier/test/test_build.py

```diff
@@ -407,6 +407,56 @@ def test_no_confirmation(self):
             self.assertFalse(os.path.isfile(fragment_path1))
             self.assertFalse(os.path.isfile(fragment_path2))
 
+    def test_keep_fragments(self):
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            setup_simple_project()
+            fragment_path1 = "foo/newsfragments/123.feature"
+            fragment_path2 = "foo/newsfragments/124.feature.rst"
+            with open(fragment_path1, "w") as f:
+                f.write("Adds levitation")
+            with open(fragment_path2, "w") as f:
+                f.write("Extends levitation")
+
+            call(["git", "init"])
+            call(["git", "config", "user.name", "user"])
+            call(["git", "config", "user.email", "user@example.com"])
+            call(["git", "add", "."])
+            call(["git", "commit", "-m", "Initial Commit"])
+
+            result = runner.invoke(_main, ["--date", "01-01-2001", "--keep"])
+
+            self.assertEqual(0, result.exit_code)
+            path = "NEWS.rst"
+            self.assertTrue(os.path.isfile(path))
```

Comment:
Maybe also have an assertion before calling the CLI that the NEWS file doesn't exists.

```suggestion
            # The NEWS file is created.
            # So this is not just `--draft`.
            self.assertTrue(os.path.isfile("NEWS.rst"))
```